File: AppSets-Android/share/src/main/java/xcj/app/share/base/DeviceName.kt
package xcj.app.share.base

import xcj.app.share.util.NameGenerator
import xcj.app.starter.util.VendorUtil

data class DeviceName(
    var rawName: String,
    var nickName: String? = null
) {
    val name: String
        get() = nickName ?: rawName

    companion object {
        val NONE: DeviceName
            get() = DeviceName("", null)

        val RANDOM: DeviceName
            get() {
                val rawName = VendorUtil.getVendorDeviceName()
                return DeviceName(rawName, NameGenerator.randomNikeName())
            }
    }
}

File: AppSets-iOS/AppSets/com/app/appsets/share/base/DeviceName.swift
//
//  DeviceName.swift
//  AppSets
//
//  Created by caiju Xu on 2025/3/3.
//

struct DeviceName: Codable {
    let rawName: String
    let nickName: String?

    var name: String {
        get {
            nickName ?? rawName
        }
    }

    public static let NONE = DeviceName(rawName: "", nickName: nil)

    public static var RANDOM: DeviceName {
        let rawName = VendorUtil.getVendorDeviceName()
        let nickName = NameGenerator.randomNikeName()
        return DeviceName(rawName: rawName, nickName: nickName)
    }
}
